lua/plugins.lua

```diff
@@ -1,193 +1,198 @@
-  },
-
-  {
-    "lewis6991/gitsigns.nvim",
-
-    config = function()
-      require("core.gitsigns").setup()
-    end,
-    event = "BufRead",
-    disable = not lvim.builtin.gitsigns.active,
-  },
-
-  -- Whichkey
-  {
-    "folke/which-key.nvim",
-    config = function()
-      require("core.which-key").setup()
-    end,
-    event = "BufWinEnter",
-    disable = not lvim.builtin.which_key.active,
-  },
-
-  -- Comments
-  {
-    "terrortylor/nvim-comment",
-    event = "BufRead",
-    config = function()
-      require("nvim_comment").setup()
-    end,
-    disable = not lvim.builtin.comment.active,
-  },
-
-  -- project.nvim
-  {
-    "ahmedkhalf/project.nvim",
-    config = function()
-      require("core.project").setup()
-    end,
-    disable = not lvim.builtin.project.active,
-  },
-
-  -- Icons
-  { "kyazdani42/nvim-web-devicons" },
-
-  -- Status Line and Bufferline
-  {
-    -- "hoob3rt/lualine.nvim",
-    "shadmansaleh/lualine.nvim",
-    -- "Lunarvim/lualine.nvim",
-    config = function()
-      require("core.lualine").setup()
-    end,
-    disable = not lvim.builtin.lualine.active,
-  },
-
-  {
-    "romgrk/barbar.nvim",
-    config = function()
-      require("core.bufferline").setup()
-    end,
-    event = "BufWinEnter",
-    disable = not lvim.builtin.bufferline.active,
-  },
-
-  -- Debugging
-  {
-    "mfussenegger/nvim-dap",
-    -- event = "BufWinEnter",
-    config = function()
-      require("core.dap").setup()
-    end,
-    disable = not lvim.builtin.dap.active,
-  },
-
-  -- Debugger management
-  {
-    "Pocco81/DAPInstall.nvim",
-    -- event = "BufWinEnter",
-    -- event = "BufRead",
-    disable = not lvim.builtin.dap.active,
-  },
-
-  -- Dashboard
-  {
-    "ChristianChiarulli/dashboard-nvim",
-    event = "BufWinEnter",
-    config = function()
-      require("core.dashboard").setup()
-    end,
-    disable = not lvim.builtin.dashboard.active,
-  },
-
-  -- Terminal
-  {
-    "akinsho/nvim-toggleterm.lua",
-    event = "BufWinEnter",
-    config = function()
-      require("core.terminal").setup()
-    end,
-    disable = not lvim.builtin.terminal.active,
-  },
-}
+local Plugins = {}
+
+function Plugins.defaults(config)
+  return {
+    -- Packer can manage itself as an optional plugin
+    { "wbthomason/packer.nvim" },
+    { "neovim/nvim-lspconfig" },
+    { "tamago324/nlsp-settings.nvim" },
+    { "jose-elias-alvarez/null-ls.nvim" },
+    { "antoinemadec/FixCursorHold.nvim" }, -- Needed while issue https://github.com/neovim/neovim/issues/12587 is still open
+    {
+      "kabouzeid/nvim-lspinstall",
+      event = "VimEnter",
+      config = function()
+        require("core.lspinstall"):config()
+      end,
+    },
+
+    { "nvim-lua/popup.nvim" },
+    { "nvim-lua/plenary.nvim" },
+    -- Telescope
+    {
+      "nvim-telescope/telescope.nvim",
+      config = function()
+        require("core.telescope"):config()
+      end,
+      disable = not config:get "telescope.active",
```

Comment:
how about creating a utility function for this? e.g: 
```lua
disable = config:condition "telescope"
``` 
also, isn't `not nil == true` ? is that behavior desirable? because all the plugins must now have the "active" attribute.

Reply:
> how about creating a utility function for this? e.g:
> 
> ```lua
> disable = config:condition "telescope"
> ```

The configuration should not share the kind of knowledge, especially if you suggest that the function should check the active field.

> also, isn't `not nil == true` ? is that behavior desirable? because all the plugins must now have the "active" attribute.

That's something I said for quite a while :)

Reply:
> The configuration should not share the kind of knowledge, especially if you suggest that the function should check the active field.

You can make it a local wrapper instead.

Reply:
I would then prefer to have some kind of factory function handle this job, but yep, we can do that